src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -133,7 +137,8 @@ void SoxSink::write(audio::Frame& frame) {
     write_(buffer_data, buffer_pos);
 }
 
-bool SoxSink::prepare_() {
+bool SoxSink::setup_buffer_() {
+    buffer_size_ = packet::ns_to_size(frame_length_, sample_rate(), channels_);
```

Comment:
Here and in SoxSource, we should also check if the calculated `buffer_size_` is zero and return false in this case.

Reply:
done.